react/components/RenderProvider.tsx

```diff
@@ -692,21 +692,28 @@ class RenderProvider extends Component<Props, RenderProviderState> {
     return assetsPromise
   }
 
-  public onLocaleSelected = (locale: string) => {
+  public onLocaleSelected = (locale: string, domain?: string) => {
     if (locale !== this.state.culture.locale) {
-      const sessionData = {
-        public: {
-          cultureInfo: {
-            value: locale,
+      const sessionData = { public: { } }
+      if(domain && domain === 'admin'){
+        sessionData.public = {
+            admin_cultureInfo: {
```

Comment:
Maybe DRY on this and declare `cultureInfo` to be reused?
```
const cultureInfo = { 
  value: locale,
}
```

Reply:
Session expects it like that. We might improve this later.